esp/bindings/http/platform/httpservice.cpp

```diff
@@ -351,6 +351,44 @@ int CEspHttpServer::processRequest()
         m_request->getEspPathInfo(stype, &pathEx, &serviceName, &methodName, false);
         ESPLOG(LogNormal,"sub service type: %s. parm: %s", getSubServiceDesc(stype), m_request->queryParamStr());
 
+        // getEspPathInfo provides all information needed to decide if the request should be
+        // traced. Create a server span, if needed, before proceding with request processing
+        // so maximize the amount of request processing that can be traced. Specifically, user
+        // authentication and authorization may generate trace output.
+        bool wantTracing = true;
+        if (!queryTraceManager().isTracingEnabled())
+            wantTracing = false;
+        else if (streq(method, GET_METHOD))
+        {
+            if (sub_serv_root == stype)
+                wantTracing = false;
+            else if (strieq(serviceName, "esp"))
+            {
+                wantTracing = !(methodName.isEmpty() ||
+                    strieq(methodName, "files") ||
+                    strieq(methodName, "xslt") ||
+                    strieq(methodName, "frame") ||
```

Comment:
is "body" deliberately missing?

How do we ensure this stays in sync with the code later in the function?  Should at least be a comment.

Reply:
Good catch. Not only is "body" not intentionally missing, but this doesn't account for the underscore that could be added to each name.

I'm thinking about defining a mapping of method name to enumerated value. This tracing check will only need to check the map for a matching entry. The subsequent handling code will be refactored to switch on the enumerated value. Only one set of string comparisons will be necessary, for a small performance boost.